### DISCUSSION TITLE
Yocto Microsemi libsocketcan

### DISCUSSION THREAD

**Discussion by Shivaraj199** [grodansparadis/vscpl2drv-socketcan#5] [General]
hello,
First of all thank you for driver code
I am trying to run a user application linux based SOC which is a RISCV architecture. My application uses socket can interfaces and apparantely the SOC I am working dosnt support the libsocketcan . I can manually load the socketcan driver in the SOC using  (As i am building the kernel with yocto). My question is it possible to us the vscpl2drv socketcan driver  cross compile it and use it in my applciation ?

**Comment by grodansparadis** [comment]
Yes that should be possible. The build is cmake based so (I have not tested) should be standard procedure as of https://cmake.org/cmake/help/book/mastering-cmake/chapter/Cross%20Compiling%20With%20CMake.html.

**Reply by Shivaraj199** [reply]
Thank you for the reply, 
vscpl2drv is is independent of hardware , I mean it is generic for all linux machine I am targeting Linux 
``
icicle-kit-es-amp 5.15.32-linux4microchip+fpga-2022.05 #1 SMP Thu May 26 12:32:36 UTC 2022 riscv64 riscv64 riscv64 GNU/Linux
``
This is the kernel of my SOC

**Comment by grodansparadis** [comment]
The driver is only dependent on socketcan and standard libs so it should work on all systems that support socketcan as far as I can tell.
